File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {
    var answer = 0;
    const arr = Array.from({length: number}, (_,index) => index + 1);
    arr.forEach(value => {
        let count = 0;
        let i = 1;
        while(i <= Math.sqrt(value)){
            
            if(value % i === 0){
                count++;
                
                if(value / i !== i){
                    count++;
                }
            }
            
            i++;
        }
        
        if(limit < count){
            answer += power;
        } else {
            answer += count
        }
    })
    return answer;
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
import math

def solution(number, limit, power):
    def count_d(n):
        cnt = 0
        for i in range(1, math.isqrt(n) + 1):
            if n % i == 0:
                cnt += 1 if i * i == n else 2
        return cnt
    answer = 0
    for i in range(1, number + 1):
        cnt = count_d(i)
        if cnt > limit:
            cnt = power
        answer += cnt
    return answer